### PR TITLE
PT-3354: Remove line items from the cart

### PR DESCRIPTION
**Pull request by eMazeika**
Description: 

9AzfePtKLA

--

QA-test: PT-3357

Demo-test: 

Download artifact URL: ghcr.io/virtocommerce/demo-xapi-app:1.0.0-pr-26-pt-3354-add3b2c8

### CODE REVIEW

**Review by Vectorfield4 — APPROVED** [review]

File: src/app/store/cart/cart.actions.ts [hunk 1]
```diff
@@ -18,6 +18,21 @@ export const getCartFailure = createAction(
...
+
+export const removeCartItemSuccess = createAction(
+  '[Cart] Remove CartItem Success',
+  props<{ lineItemId: string }>()
+);
+
+export const removeCartItemFailure = createAction(
+  '[Cart] Remove CartItem Failure',
+  props<{ error: ApolloError }>()
+);
```
Comment (ksavosteev) [inline src/app/store/cart/cart.actions.ts]: can be removed if not used

File: src/app/modules/checkout/components/line-items/line-items.component.ts [hunk 2]
```diff
@@ -12,11 +15,64 @@ import { Cart } from 'src/app/models/cart.model';
...
+          break;
+        }
+
+        return {
+          name: p?.name,
+          displayValue,
+        };
+      });
+  }
+
```
Comment (asvishnyakov) [inline src/app/modules/checkout/components/line-items/line-items.component.ts:61]: Shouldn't we introduce something like service for dynamic properties conversion?
Also, may be it possible to return typed value for dynamic property? Or introduce special type for them. Something like
```typescript
type DP = { valueType: 'Integer', value: number } | { valueType: 'DateTime', value: Date }
```

(just to think about)

**Review by asvishnyakov — APPROVED** [review]

**Comment by asvishnyakov** [comment]
It seems like this PR and https://github.com/VirtoCommerce/vc-demo-xapi-app/pull/24 has shared code or one include another. Is it ok?

File: src/app/modules/checkout/components/line-items/line-items.component.ts (hunk 2 above)
Comment (eMazeika) [reply src/app/modules/checkout/components/line-items/line-items.component.ts:61]: TODO: add dynamic property component containing all the logic. But only when the code would be reused in yet another place (avoid duplication).

**Pull request merged**